Submitted By: Jim Gifford (patches at jg555 dot com)
Date: 2006-04-21
Initial Package Version: 3.4.6
Origin: Jim Gifford
Upstream Status: Delayed till {unknown}
Description: Fixes demangle static constraints

diff -Naur gcc-3.4.6.orig/libiberty/cp-demangle.h gcc-3.4.6/libiberty/cp-demangle.h
--- gcc-3.4.6.orig/libiberty/cp-demangle.h	2004-02-25 10:41:51.000000000 -0800
+++ gcc-3.4.6/libiberty/cp-demangle.h	2006-04-22 06:59:32.000000000 -0700
@@ -131,18 +131,27 @@
 
 /* Functions and arrays in cp-demangle.c which are referenced by
    functions in cp-demint.c.  */
+#ifdef IN_GLIBCPP_V3
+#define CP_STATIC_IF_GLIBCPP_V3 static
+#else
+#define CP_STATIC_IF_GLIBCPP_V3 extern
+#endif
 
-extern const struct demangle_operator_info cplus_demangle_operators[];
+CP_STATIC_IF_GLIBCPP_V3
+const struct demangle_operator_info cplus_demangle_operators[];
 
 #define D_BUILTIN_TYPE_COUNT (26)
 
-extern const struct demangle_builtin_type_info
+CP_STATIC_IF_GLIBCPP_V3
+const struct demangle_builtin_type_info
 cplus_demangle_builtin_types[D_BUILTIN_TYPE_COUNT];
 
-extern struct demangle_component *
+CP_STATIC_IF_GLIBCPP_V3
+struct demangle_component *
 cplus_demangle_mangled_name PARAMS ((struct d_info *, int));
 
-extern struct demangle_component *
+CP_STATIC_IF_GLIBCPP_V3
+struct demangle_component *
 cplus_demangle_type PARAMS ((struct d_info *));
 
 extern void
